src/app/setup-wizard/pre-visit/pre-assessment-setup/pre-assessment-setup.component.ts

```diff
@@ -101,7 +105,29 @@ export class PreAssessmentSetupComponent {
       this.energyEquipmentOptions = equipments;
     });
 
-    this.companyEnergyUnit = this.companyIdbService.getByGUID(this.onSiteVisit.companyId).companyEnergyUnit;
+    this.companySub = this.companyIdbService.selectedCompany.subscribe(_company => {
+      if (_company.guid === this.onSiteVisit.companyId) {
```

Comment:
The If else statement here isn't necessary. The selected company and facility are set when the wizard is loaded in and should always match the selected onsite visit. If that's not the case we can look at how those two Behavior Subjects are set.

Reply:
thanks, removed